.github/workflows/main.yml

```diff
@@ -0,0 +1,75 @@
+name: Build and Upload JAR
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'src/**'
+      - 'build.gradle'
+      - '.github/workflows/main.yml'
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'src/**'
+      - 'build.gradle'
+      - '.github/workflows/main.yml'
+
+env:
+  GRADLE_OPTS: -Dorg.gradle.daemon=false
+  DO_NOT_TRACK: true
```

Comment:
Don't think that's needed here, you pass it to the gradlew build step specifically already